client/client.go

```diff
@@ -219,36 +220,37 @@ func NewClientWithPort(port string) (client Client, err error) {
 // NewClientWithAddress instantiates Dapr using specific address (including port).
 func NewClientWithAddress(address string) (client Client, err error) {
 	if address == "" {
-		return nil, errors.New("nil address")
+		return nil, errors.New("empty address")
 	}
 	logger.Printf("dapr client initializing for: %s", address)
 
 	timeoutSeconds, err := getClientTimeoutSeconds()
 	if err != nil {
 		return nil, err
 	}
-	ctx, ctxCancel := context.WithTimeout(context.Background(), time.Duration(timeoutSeconds)*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeoutSeconds)*time.Second)
 	conn, err := grpc.DialContext(
 		ctx,
 		address,
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithUserAgent("dapr-sdk-go/"+version.SDKVersion),
 		grpc.WithBlock(),
 	)
+	cancel()
```

Comment:
Working on this, I found a bug here. There's no need to save the context because it's only used by the `DialContext` method, and it's expired after the timeout anyways. So, I've removed saving the context in the object